component-samples/components-oauth/main.js

```diff
@@ -0,0 +1,57 @@
+/* Copyright 2024 Esri
+ *
+ * Licensed under the Apache License Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import "./styles.css";
+
+// Import the ArcGIS Map Components
+import "@arcgis/map-components/dist/components/arcgis-map";
+import "@arcgis/map-components/dist/components/arcgis-legend";
+import "@arcgis/map-components/dist/components/arcgis-search";
+
+// Import ArcGIS core modules for OAuth authentication
+import esriConfig from "@arcgis/core/config";
+import esriId from "@arcgis/core/identity/IdentityManager.js";
+import OAuthInfo from "@arcgis/core/identity/OAuthInfo";
+
+// Configure the portal URL and OAuth settings
+esriConfig.portalUrl = "YOUR_PORTAL_URL"; // Replace with your actual Portal URL
+
+const oAuthInfo = new OAuthInfo({
+    appId: "YOUR_APP_ID", // Replace with your actual App ID
+    portalUrl: esriConfig.portalUrl,
+    popup: false,
+});
+
+esriId.registerOAuthInfos([oAuthInfo]);
+
+// Function to load map components after authentication
+function loadMapComponents() {
+    const mapElement = document.querySelector('arcgis-map');
+    mapElement.itemId = "d5dda743788a4b0688fe48f43ae7beb9";
```

Comment:
Does the item id need to be set here and in the html?

Reply:
I missed this. I think it's to wait for the auth before assigning an itemid that needs it. In that case, one of two things should be done.

1. No `item-id` in html element
2. Dynamically import map-comps modules in `loadComponents` - `await Promise.allResolve([import(), ...]);`

Reply:
The empty `item-id` will let the default basemap load, and if you update the item-id after authenticating, the map would update and have all the needed auth info to do what it needs to do.

Reply:
The dynamic imports should leave a blank screen until auth workflow is done and then load.

Reply:
`esriId.checkSignInStatus` doesn't force a sign in. Maybe it'd be better to say that if `item-id` is set to a secure item or includes secure items, then the user will be prompted to sign in.

Reply:
I see `getCredential` is also called, but it does't seem like either need to be called if we instead relied on it to be handled automatically.

Reply:
I would load the comps like this:
```js
import "./styles.css";

// Import ArcGIS core modules for OAuth authentication
import esriConfig from "@arcgis/core/config";
import esriId from "@arcgis/core/identity/IdentityManager.js";
import OAuthInfo from "@arcgis/core/identity/OAuthInfo";

// Configure the portal URL and OAuth settings
esriConfig.portalUrl = "YOUR_PORTAL_URL"; // Replace with your actual Portal URL

const oAuthInfo = new OAuthInfo({
    appId: "YOUR_APP_ID", // Replace with your actual App ID
    portalUrl: esriConfig.portalUrl,
    popup: false,
});

esriId.registerOAuthInfos([oAuthInfo]);

const mapElement = document.querySelector('arcgis-map');

mapElement.addEventListener('arcgisViewReadyChange', event => {
    console.log('MapView ready', event);
});

// Function to load map components after authentication
async function loadMapComponents() {
    await Promise.allSettled([
        // Import the ArcGIS Map Components after OAuth workflow
        import("@arcgis/map-components/dist/components/arcgis-map"),
        import("@arcgis/map-components/dist/components/arcgis-legend"),
        import("@arcgis/map-components/dist/components/arcgis-search"),
    ]);
}
...
```

You can add the event listener before the components are loaded.

Reply:
And leave the `item-id` in the HTML element and don't update in JS. This is closer to what a user is probably going to do.

Reply:
Updated the sample accordingly and removed forced sign-in and dynamic imports to reduce complexity